### PR TITLE
OBPIH-4705 Location functionality

### PR DESCRIPTION
**Pull request by drodzewicz**

### CODE REVIEW

File: src/js/components/location/LocationButton.jsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
+
+import PropTypes from 'prop-types';
+import { RiMapPinLine } from 'react-icons/ri';
+
+const LocationButton = ({ location, onClick }) => {
+  const [isHovered, setIsHovered] = useState(false);
+
+  const buttonStyle = () => {
+    if (!location.backgroundColor || ['FFFFFF', 'FFFF'].includes(location.backgroundColor)) {
```
Comment (drodzewicz) [inline src/js/components/location/LocationButton.jsx:10]: For locations inside a modal there is a feature that applies a color on **LocationButton** based on the `location.backgroundColor` value 
image

The problem was that by default most of the locations have a white color _(and not a consistent value like `#FFFFFF` and sometimes `#FFFF`)_

File: src/js/components/Layout/Breadcrumbs.jsx [hunk 2]
```diff
@@ -1,65 +1,48 @@
...
+    <div className="breadcrumbs-container d-flex">
+      <a className="item-breadcrumbs" href="/openboxes">
+        <img alt="Breadcrumbs" src="/openboxes/images/skin/house.png" />
+      </a>
+      <img className="item-breadcrumbs" alt="/" src="/openboxes/images/bc_separator.png" />
+      <a
+        role="button"
+        href="#"
+        className="item-breadcrumbs"
+      > {currentLocationName} test
```
Comment (awalkowiak) [inline src/js/components/Layout/Breadcrumbs.jsx]: `test`

File: src/css/colors.scss [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+$blue-200: #F0F6FF;
+$blue-201: #F4F7FE;
+$blue-300: #E9EDF4;
+$blue-400: #7AA3F3;
+$blue-500: #2264E5;
+$blue-700: #2B3674;
+$blue-800: #202750;
+
+// GRAY
+$gray-100: #EAEAE;
```
Comment (awalkowiak) [inline src/css/colors.scss:45]: Is it missing `A` at the end?

File: src/css/main.scss [hunk 4]
```diff
@@ -851,61 +829,16 @@ div.crossed-out .table-inner-row:before {
...
-.tabs-panel-container {
-  border: 1px solid $gray-gamma;
-  border-radius: 0 5px 5px 0;
-  border-left: none;
-  padding: 10px 12px;
-  flex: 1 1 auto;
-}
+/* LOCATION CHOOSER
+----------------------------*/
+@import '../js/components/location/LocationChooser';
```
Comment (awalkowiak) [inline src/css/main.scss:841]: Is this required here?

File: src/js/components/location/LocationChooser/LocationChooserButton/LocationChooserButton.jsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+
+import PropTypes from 'prop-types';
+import { Tooltip } from 'react-tippy';
+
+import Translate from 'utils/Translate';
+
+import './LocationChooserButton.scss';
+
```
Comment (kchelstowski) [inline src/js/components/location/LocationChooser/LocationChooserButton/LocationChooserButton.jsx:9]: shouldn't we use absolute imports everywhere? e.g. `components/LocationChooserButton/LocationChooseButton.scss`

File: src/js/components/location/LocationChooser/LocationChooserModal/LocationChooserModal.jsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+import PropTypes from 'prop-types';
+import { RiCloseLine } from 'react-icons/ri';
+import Modal from 'react-modal';
+import { Tab, TabList, TabPanel, Tabs } from 'react-tabs';
+
+import Spinner from 'components/spinner/Spinner';
+import Translate from 'utils/Translate';
+
+import LocationButton from './LocationButton';
+
```
Comment (kchelstowski) [inline src/js/components/location/LocationChooser/LocationChooserModal/LocationChooserModal.jsx:12]: same as above

**Review by kchelstowski — CHANGES_REQUESTED** [review]

File: src/js/components/location/LocationChooser/index.jsx [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+        * */
+        const locationsGroupedByOrganization = _.groupBy(data, location => _.get(location, 'organizationName'));
+        const locOrg = Object.entries(locationsGroupedByOrganization)
+          .map(([organization, locations]) => {
+            const locationsGroupedByGroup = _.groupBy(locations, location => _.get(location, 'locationGroup.name', null));
+            const groups = Object.entries(locationsGroupedByGroup)
+              .map(([group, locationList]) => ({ group, locations: locationList }))
+              .sort((a, b) => (a.group > b.group ? 1 : -1));
+            return { organization, groups };
+          })
```
[the remark this answers is not in the document]
Comment (drodzewicz) [reply src/js/components/location/LocationChooser/index.jsx]: The `apiClient` utils functions has already implemented catching for bad requests and displays an Alert with an error message.

**Review by kchelstowski — APPROVED** [review]

**Review by awalkowiak — APPROVED** [review]

**Pull request merged**